builder/src/lib.rs

```diff
@@ -56,40 +56,70 @@ impl Service {
+        info!(deployment_id, "extracted the tar archive content");
+
         let secrets = get_secrets(path).await?;
         build_flake_file(path)?;
+        info!(deployment_id, "created the project flake file successfully");
 
-        let mut cmd = Command::new("nix");
         let output_path = path.join("_archive");
-        cmd.args([
-            "build",
-            "--no-write-lock-file",
-            "--impure",
-            "--log-format",
-            "bar-with-logs",
-            "--out-link",
-            output_path.to_str().unwrap(),
-            path.to_str().unwrap(),
-        ])
-        .stdout(Stdio::piped());
-
-        let mut child = cmd.spawn()?;
-        let stdout = child.stdout.take().expect("to get handle on stdout");
-
-        let mut reader = BufReader::new(stdout).lines();
-
-        let id = deployment_id.clone();
-        tokio::spawn(async move {
-            while let Some(line) = reader.next_line().await.expect("to get line") {
-                info!(deployment_id = %id, "{line}");
+        let mut child = Command::new("nix")
+            .args([
+                "build",
+                "--no-write-lock-file",
+                "--impure",
+                "--log-format",
+                "bar-with-logs",
+                "--out-link",
+                output_path.to_str().unwrap(),
+                path.to_str().unwrap(),
+            ])
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            .spawn()?;
+
+        let stdout = child.stdout.take().expect("to get handle on stderr");
+        let stderr = child.stderr.take().expect("to get handle on stderr");
+        let mut stderr_reader = BufReader::new(stderr).lines();
+        let mut stdout_reader = BufReader::new(stdout).lines();
+        let mut stdout_ended = false;
+        let mut stderr_ended = false;
+        while !(stdout_ended && stderr_ended) {
+            tokio::select! {
+                stderr_line = stderr_reader.next_line() => {
+                    match stderr_line {
+                        Ok(Some(line)) => info!(deployment_id, "{line}"),
+                        Ok(None) => stderr_ended = true,
+                        Err(err) => {
+                            error!(deployment_id, "unexpected stderr stream close: {}", err);
+                            stderr_ended = true;
+                        }
+                    }
+                },
+                stdout_line = stdout_reader.next_line() => {
+                    match stdout_line {
+                        Ok(Some(line)) => info!(deployment_id, "{line}"),
+                        Ok(None) => stdout_ended = true,
+                        Err(err) => {
+                            error!(deployment_id, "unexpected stdout stream close: {}", err);
+                            stdout_ended = true;
+                        }
+                    }
+                }
             }
-        });
+        }
```

Comment:
So this is to combine stderr and stdout, right? Could we use the same strategy we use in the [old builder](https://github.com/shuttle-hq/shuttle/blob/main/service/src/builder.rs#L248-L266)?

Reply:
Open a ticket for this, will cover it in a next PR.